File: Sorting/boj_1931.py
n = int(input())

time = [[0, 0] for _ in range(n)]
for i in range(n):
	s, e = map(int, input().split())
	time[i][0] = s
	time[i][1] = e

time.sort(key=lambda x : (x[1], x[0]))

end_time = time[0][1]
cnt = 1
for i in range(1, n):
	if time[i][0] >= end_time:
		cnt += 1
		end_time = time[i][1]

print(cnt)

File: Greedy/boj_1931.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int n;

    cin >> n;

    vector<pair<int, int>> time;

    while (n--) {
        int start, end;

        cin >> start >> end;

        time.emplace_back(end, start);
    }

    sort(time.begin(), time.end());

    int end = time[0].first, cnt = 1;

    for (int i = 1; i < time.size(); i++) {
        if (time[i].second >= end) {
            end = time[i].first;
            cnt++;
        }
    }

    cout << cnt;

    return 0;
}
